syntax = "proto3";

package transaction_verification;

import "google/protobuf/empty.proto";

service TransactionVerificationService {
    rpc InitOrder (InitializationRequest) returns (google.protobuf.Empty) {}
    rpc VerifyItems (ContinuationRequest) returns (VerificationResponse) {}
    rpc VerifyUserData (ContinuationRequest) returns (VerificationResponse) {}
    rpc VerifyCreditCard (ContinuationRequest) returns (VerificationResponse) {}
}

message InitializationRequest {
    string order_id = 1;
    User user = 2;
    repeated Item items = 3;
    CreditCard credit_card = 4;
}

message ContinuationRequest {
    string order_id = 1;
}

message VerificationResponse {
    bool verified = 1;
}

message User {
    string name = 1;
    string contact = 2;
    Address address = 3;
}

message Address {
    string street = 1;
    string city = 2;
    string state = 3;
    string zip = 4;
    string country = 5;
}

message Item {
    string name = 1;
    int32 quantity = 2;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}
